### PR TITLE
#12919: Cleanup 16 Unary Backward ops

### PR DESCRIPTION
**Pull request by VirdhatchaniKN**
Issue : #12919
[CI](https://github.com/tenstorrent/tt-metal/actions/runs/10991084972) : PASSED
Test Files : 

- reciprocal_bw : 
  - `tests/ttnn/unit_tests/operations/backward/complex_ops/test_backward_recip.py` - PASSED
  - `tests/ttnn/unit_tests/operations/backward/test_backward_reciprocal.py` - PASSED
- fill_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_fill.py` - PASSED
- pow_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_pow.py` - PASSED
- rpow_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_rpow.py` - PASSED
- div_no_nan_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_div_no_nan.py` - PASSED
- polygamma_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_polygamma.py` - PASSED
- multigammaln_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_mvlgamma.py` - PASSED
- lgamma_bw : `/tests/ttnn/unit_tests/operations/backward/test_backward_lgamma.py` - PASSED
- hardsigmoid_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_hardsigmoid.py` - PASSED
- cos_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_cos.py` - PASSED
- acosh_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_acosh.py` - PASSED
- relu_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_relu.py` - PASSED
- logit_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_logit.py` - PASSED
- floor_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_floor.py` - PASSED
- round_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_round.py` - PASSED
- log_bw : `tests/ttnn/unit_tests/operations/backward/test_backward_log.py` - PASSED

### CODE REVIEW

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by TT-BrianLiu — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp [hunk 1]
```diff
@@ -37,39 +37,29 @@ struct ExecuteUnaryBackwardThreshold {
...
-        const Tensor &input_tensor_a_arg,
-        const Tensor &input_tensor_b_arg,
-        const std::optional<MemoryConfig> &memory_config = std::nullopt) {
-        auto output_memory_config = memory_config.value_or(input_tensor_a_arg.memory_config());
-        return OpHandler<unary_backward_op_type>::handle(grad_tensor_arg, input_tensor_a_arg, input_tensor_b_arg, output_memory_config);
-        }
-
+#define DEFINE_UNARY_BACKWARD_OPERATION_WITH_FLOAT(op_name) \
+struct ExecuteUnaryBackward##op_name { \
+    static std::vector<Tensor> invoke( \
```
Comment (eyonland) [inline ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp:42]: Do we need a macro here?  Can we not just pass the op_name as a template argument?

Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp]: Will Update the code

Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp]: I've addressed your comment

Comment (VirdhatchaniKN) [reply ttnn/cpp/ttnn/operations/eltwise/unary_backward/unary_backward.hpp]: I've addressed your comment 
https://github.com/tenstorrent/tt-metal/actions/runs/11012896880 - passed

**Review by bharane-ab — APPROVED** [review]

**Comment by VirdhatchaniKN** [comment]
Closing PR due to function changes

**Pull request closed without merging**